### DISCUSSION TITLE
How do you get whitespace from page.chars?

### DISCUSSION THREAD

**Discussion by inf3rnus** [jsvine/pdfplumber#465] [Q&A]
Hi there again, not sure if I'm doing something stupid, but how do I extract whitespace characters from the page object?

Best,
Aaron

**Comment by jsvine** [comment]
Hiya! If you’re not seeing whitespace characters in `page.chars`, that’s typically because the software that generated the PDF did not include them but instead imitated whitespace by adding distance between the non-whitespace characters. As such, identifying “whitespace” ends up being a sort of mathematical exercise that varies from PDF to PDF but can typically be determined by calculating the difference between `x0` of a char and `x1` of the char that preceded it (assuming both chars are at the same y-position).
